reader.js

```diff
@@ -54,90 +61,111 @@ function respondToHTTPReq(request, response) {
-                ids[measure.id] = measure.time;
-            }
-        });
 
+        if (result !== null) {
+            result.forEach(measure => {
+                if (!ids.hasOwnProperty(measure.id)) {
+                    ids[measure.id] = measure.time._nanoISO;
+                }
+            });
+        }
         response.end(JSON.stringify(ids));
 
     }).catch(function (err) {
         console.error(`Error querying data from InfluxDB! ${err.stack}`);
+        console.error(err);
 
         response.writeHead(500);
-        response.end(JSON.stringify({error: err.message}));;
+        response.end(JSON.stringify({error: err.message}));
     });
 }
 
-function respondToTransmissionsQuery(mission, ids, response) {
-    let influx;
+function respondToTransmissionsQuery(mission, timestamps, response) {  // by time instead of id
+
+    influxConnection.then(influx => {
 
-    influxConnection.then(influxd => {
-        influx = influxd;
-        return influx.getMeasurements();
+        if (keys.length === 0) {
+            response.end(JSON.stringify({}));
+            return null;
+        }
 
-    }).then(names => {
         let queries = [];
-        ids.forEach(id => {
+        timestamps.forEach(timestamp => {
 
-            if (!/^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i.test(id)) {
-                response.writeHead(400);
-                response.end('Wrong ids provided');
-                return;
-            }
 
-            let query = `select * from`;
-            names.forEach(name => {
-                query += ` ${name},`;
-            });
-            query = query.slice(0, -1);
-            query += ` where (mission = '${mission}') and (id = '${id}')`;
-            queries.push(query);
+            let query = `select * from `;
+            let namesString = keys.join(',');
+            query += namesString;
+            query += ` where mission = '${mission}' and time >= '${timestamp.substring(0,timestamp.search(','))}' and time <= '${timestamp.substring(timestamp.search(',')+1)}'`;
+            console.log(query);
 
+            queries.push(query);
         });
 
         return influx.query(queries);
 
     }).then(result => {
+        let transmissions = {};
 
-        let transmissions = [];
-        result.forEach(point => {
-            let transmission = {};
-            let names = point.groupRows.map((groupRow) => groupRow.name);
-            point.forEach(measurement => {
-                transmission[names.shift()] = measurement.value;
+        if (Array.isArray(result[0])) {
+            result.forEach(timeGroup => {
+                timeGroup.groupRows.forEach((group) => {
+                    const name = group.name;
+
+                    group.rows.forEach((point) => {
+                        if (!transmissions[point.id]) {
+                            transmissions[point.id] = {'Human Time':point.time._nanoISO, mission : Number(mission)};
```

Comment:
We also need to put in a `timestamp` property (in ms since unix epoch, the same sort of thing you'd get from `Date.now()`) so that the computer can read it.